hie-plugin-api/Haskell/Ide/Engine/Config.hs

```diff
@@ -46,19 +58,20 @@ instance FromJSON Config where
 --                                                                                          ,("maxNumberOfProblems",Number 100.0)]))])}}
 
 instance ToJSON Config where
-  toJSON (Config h m l c) = object [ "languageServerHaskell" .= r ]
+  toJSON (Config h m l c f) = object [ "languageServerHaskell" .= r ]
     where
       r = object [ "hlintOn"              .= h
                  , "maxNumberOfProblems"  .= m
                  , "liquidOn"             .= l
                  , "completionSnippetsOn" .= c
+                 , "formatOnImportOn"     .= f
                  ]
 
 -- ---------------------------------------------------------------------
 
 -- | For the diagnostic providers in the config, return a map of
 -- current enabled state, indexed by the plugin id.
-getDiagnosticProvidersConfig :: Config -> Map.Map PluginId Bool
+getDiagnosticProvidersConfig :: Config -> Map.Map T.Text Bool
```

Comment:
Why is this different?

Reply:
`PluginId` is defined inside `PluginsIdeMonads`, which now imports `Config` so its to avoid a circular dependency. This is probably a sign that we should split the module up a bit

Reply:
Ah, makes sense

Reply:
Should we not keep it as a PluginId, just define it in a different place? I am not all that happy with it sudennly being `Text`, we are losing coherence.

Reply:
Can this function be moved into `LspStdio.hs`? to avoid the circular dependency? It feels like it belongs in `Config.hs` though. I couldn't really find a logical grouping of Plugin related types to move out `PluginsIdeMonads` since they all reference IdeGhcM etc.
Another possibility is that `PluginDescriptor` gets moved into `PluginsIdeMonads` where the actual PluginDescriptor type lives